_data/data.js

```diff
@@ -239,14 +241,25 @@ const downloadImage = async function (url, file) {
     });
 
     response.data.on('end', () => {
-      console.log(`Image received: ${path} (${hash.digest('hex')})`);
+      hashCode = hash.digest('hex');
+      console.log(`Image received: ${path} (${hashCode})`);
     });
     writer.on('finish', () => {
       console.timeEnd(label);
+      const hashFileName = `${hashCode}${extension}`;
+      const newPath = Path.join(IMAGE_DIRECTORY, hashFileName);
+      const newVirtualPath = Path.join(IMAGE_PATH, hashFileName);
+
```

Comment:
You can add the cache checker here. If the file with the same hash and extension already exists, you don't need to call `FS.renameSync` and can assume it's the same file.